### PR TITLE
Sync email for delegates fix

### PR DESCRIPTION
**Pull request by danieljames-dj**

### CODE REVIEW

**Comment by gregorbg** [comment]
I am not a big fan of including child roles for every group in every case by default.

`def roles` should only be direct roles.
You can add a second `def child_roles` and then combine them at the place where the methods are being called, if needed.

**Comment by danieljames-dj** [comment]
> I am not a big fan of including child roles for every group in every case by default.
> 
> `def roles` should only be direct roles. You can add a second `def child_roles` and then combine them at the place where the methods are being called, if needed.

Done

File: WcaOnRails/app/models/user_group.rb [hunk 1]
```diff
@@ -60,10 +60,18 @@ def roles
     role_list
   end
 
+  def child_roles
+    child_groups.map(&:roles).flatten
```
Comment (gregorbg) [inline WcaOnRails/app/models/user_group.rb:64]: `child_groups.flat_map(&:roles)` 😉

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: WcaOnRails/app/models/user_group.rb (hunk 1 above)
Comment (danieljames-dj) [reply WcaOnRails/app/models/user_group.rb]: Done

**Review by gregorbg — APPROVED** [review]

**Pull request merged**